api/v1alpha1/search_types.go

```diff
@@ -0,0 +1,199 @@
+
+	// +optional
+	// Kubernetes secret name containing user provided db secret
+	// Secret contains connection url, certificates
+	CustomDbConfig string `json:"customDbConfig,omitempty"`
+}
+
+type SearchDeployments struct {
+	// +optional
+	// Configuration for DB Deployment
+	Database DeploymentConfig `json:"database,omitempty"`
+
+	// +optional
+	// Configuration for indexer Deployment
+	Indexer DeploymentConfig `json:"indexer,omitempty"`
+
+	// +optional
+	// Configuration for collector Deployment
+	Collector DeploymentConfig `json:"collector,omitempty"`
+
+	// +optional
+	// Configuration for api Deployment
+	API DeploymentConfig `json:"api,omitempty"`
+
+	// +optional
+	// Configuration for addon installed collector Deployment
+	RemoteCollector DeploymentConfig `json:"remoteCollector,omitempty"`
+}
+
+type DeploymentConfig struct {
+	// +optional
+	// Number of pod instances for deployment
+	ReplicaCount int `json:"replicaCount,omitempty"`
+	// +optional
+	// Compute Resources required by deployment
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+	//Image_override
+	ImageOverride string `json:"imageOverride,omitempty"`
+
+	// +optional
+	//ImagePullSecret
+	ImagePullSecret string `json:"imagePullSecret,omitempty"`
+
+	//ImagePullPolicy
+	// +optional
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+
+	// NodeSelector to schedule on nodes with matching labels
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	//Proxy config , if remote collectors need to override
+	// +optional
+	ProxyConfig map[string]string `json:"proxyConfig,omitempty"`
+}
+
+type StorageSpec struct {
+	// +optional
+	// name of the storage class
+	StorageClassName string `json:"storageClassName,omitempty"`
+	// +optional
+	// storage capacity
+	Size *resource.Quantity `json:"size,omitempty"`
+}
+
+type FilterSpec struct {
+	// +optional
+	// Allowed resources from collector
+	AllowedResources ResourceListSpec `json:"allowedResources,omitempty"`
+
+	// +optional
+	// Denied resources from collector
+	DeniedResources ResourceListSpec `json:"deniedResources,omitempty"`
+}
+
+type ResourceListSpec struct {
+	//API Group names to be filtered
+	APIGroups []string `json:"apiGroups,omitempty"`
+	//Resource names to be filtered
+	Resources []string `json:"resources,omitempty"`
+	// +optional
+	//Cluster Labels this filter to be applied
+	ClusterLabels []string `json:"clusterLabels,omitempty"`
+}
+
+// SearchStatus defines the observed state of Search
+type SearchStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	// Human readable health state
+	SearchHealth string `json:"searchHealth,omitempty"`
+	// Database used by search
+	DBInUse string `json:"dbInUse,omitempty"`
+	// Storage used by database
+	StorageInUse string `json:"storageInUse,omitempty"`
+	// +optional
```

Comment:
Maybe `InUse` is implied and not needed because status represents current state.